src/foundry/foundry.js/clientDocuments/combat.d.ts

```diff
@@ -16,7 +18,202 @@ declare global {
+
+    /** Begin the combat encounter, advancing to round 1 and turn 1 */
+    startCombat(): Promise<this>;
+
+    /**
+     * Define how the array of Combatants is sorted in the displayed list of the tracker.
+     * This method can be overridden by a system or module which needs to display combatants in an alternative order.
+     * By default sort by initiative, next falling back to name, lastly tie-breaking by combatant id.
+     */
+    protected _sortCombatants(
+      a: InstanceType<ConfiguredDocumentClass<typeof Combatant>>,
+      b: InstanceType<ConfiguredDocumentClass<typeof Combatant>>
+    ): number;
+
+    protected _onCreate(data: this['data']['_source'], options: DocumentModificationOptions, userId: string): void;
+
+    protected _onUpdate(
+      changed: DeepPartial<this['data']['_source']> & Record<string, unknown>,
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    protected _onDelete(options: DocumentModificationOptions, userId: string): void;
+
+    protected _onCreateEmbeddedDocuments(
+      embeddedName: string,
+      documents: foundry.abstract.Document<any, any>[],
+      result: Record<string, unknown>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    protected _onUpdateEmbeddedDocuments(
+      embeddedName: string,
+      documents: foundry.abstract.Document<any, any>[],
+      result: Record<string, unknown>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    protected _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: foundry.abstract.Document<any, any>[],
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+
+    /** @deprecated since 0.8.0 */
+    getCombatant(id: string): this['combatants']['get'];
+
+    /** @deprecated since 0.8.0 */
+    createCombatant(
+      data: Array<Record<string, unknown>>,
+      context?: DocumentModificationContext
+    ): this['createEmbeddedDocuments'];
+
+    /** @deprecated since 0.8.0 */
+    updateCombatant(
+      data: DeepPartial<Combatant['data']> & Record<string, unknown>,
+      context?: DocumentModificationContext
+    ): Combatant['update'];
+
+    /** @deprecated since 0.8.0 */
+    deleteCombatant(id: string, options?: DocumentModificationContext): Combatant['delete'];
   }
 }
 
+interface InitiativeOptions {
+  /**
+   * A non-default initiative formula to roll. Otherwise the system default is used.
+   * @defaultValue `null`
+   */
+  formula?: string | null;
+  /**
+   * Update the Combat turn after adding new initiative scores to keep the turn on the same Combatant.
+   * @defaultValue `true`
+   */
+  updateTurn?: boolean;
+  /**
+   * Additional options with which to customize created Chat Messages
+   * @defaultValue `{}`
+   */
+  messageOptions?: object; //TODO Type properly once ChatMessage is typed
+}
+
+interface RoundData {
```

Comment:
Why are the members of this interface optional? aren't they always `null` in foundry, if they aren't given?